runtime/list.go

```diff
@@ -163,6 +163,14 @@ func listAppend(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return None, nil
 }
 
+func listExtend(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	argc := len(args)
+	if argc != 2 {
+		return nil, f.RaiseType(TypeErrorType, fmt.Sprintf("extend() takes exactly one argument (%d given)", argc-1))
```

Comment:
If listExtend gets called with empty args (e.g. `list.extend()`), won't this message say "extend() takes exactly one argument (-1 given)"? CPython has a `method` type (distinct from `instancemethod`) that gives this error in that case:

`TypeError: descriptor 'extend' of 'list' object needs an argument`

I don't think we need to worry about implementing `method` right now. Maybe we should settle for having the error message be:

`fmt.Sprintf("extend() takes exactly two arguments (%d given)", argc)`

It's a little weird, but I think less confusing than saying "(-1 given)" in some cases.